polaris.shopify.com/src/components/Markdown/components/Colors/index.tsx

```diff
@@ -16,39 +18,178 @@ const colors = colorsObj as unknown as Colors;
 
 export function Colors() {
   const colorOrder = [
-    'gray',
+    'red',
+    'orange',
+    'yellow',
+    'lime',
     'green',
+    'cyan',
     'teal',
+    'azure',
     'blue',
     'purple',
-    'red',
-    'orange',
-    'yellow',
+    'magenta',
+    'rose',
   ];
 
+  let a11ySelectColorOne: boolean = false;
+  interface a11yColor {
+    r: string;
+    g: string;
+    b: string;
+    a: string;
+    id: string;
+  }
+  let a11yColorOne: a11yColor;
+  let a11yColorTwo: a11yColor;
+  let a11yRatio: number = 0;
+  const selectColor = (colorValue: string, colorName: string) => {
+    a11ySelectColorOne = !a11ySelectColorOne;
+    let rgbaArray = colorValue.replace(/[^\d,]+/g, '').split(',');
+
+    document.querySelectorAll('.' + styles.ColorsSwatch).forEach((element) => {
```

Comment:
for a case like this color palette we should not be reaching into the DOM directly as this will cause discrepencies with the react virtual dom when things re-render. This could lead to errors like the one above. Instead, can we break things out into react components and useState to handle the interactivity of this component?